### PR TITLE
Rg/db elasticpool feature

### PR DESCRIPTION
**Pull request by gkiranreddy**
- Sync dbs - elastic pool and associated db sync feature
- Custom fields 
- Sync Servers - app service feature with custom fields

### CODE REVIEW

**Review by dcolebatch — CHANGES_REQUESTED** [review]
Hi @gkiranreddy - please take a look at this feedback

File: azure_dbs.rb [hunk 1]
```diff
@@ -238,20 +254,58 @@ def list_databases_by_server(subscription, resource_group, server_name)
...
 class DBFetcher
   extend AzureDB
   extend AzureDBServer
   extend AzureHelper
+  extend AzureElasticPool
 
   def self.sync_to_tidal_portal(data, type)
     file_path = "/tmp/tidal_#{type}_data.json"
+    # Temporary file creation for Windows Run
+    # file_path = "#{Dir.pwd}/tidal_#{type}_data.json"
```
[the remark this answers is not in the document]
Comment (gkiranreddy) [reply azure_dbs.rb:297]: I will make this change

**Review by gkiranreddy — COMMENTED** [review]
- I updated code based on the review, 
- And also added batching feature for VMs script, Currently I set the SAVE_INTERNAL to 10 just to test. But this can be updated to what ever we see fit

**Comment by dcolebatch** [comment]
Thanks @gkiranreddy!

1) Just confirming: you ran this and it worked ok?  Particularly the custom fields changes, did they come through as you expected?

2) with respect to the periodic saving to file- I believe the intent here was to be able to handle very large environments without running out of memory, or causing 500s on the Tidal API?

 I have two observations:
a) We appear to write to file, but also maintain the data in a global variable? (So, our memory footprint might actually be worse now)
b) we delete the file at the end, which would prevent us from being able to re-play the data to `tidal sync...` manually when troubleshooting.

I wonder if we might consider periodically writing to file *and* syncing to tidal tools if some flag is set?

e.g. the below, with no STDOUT writing of JSON:
```./azure_vms.rb -sync-page 10```

as an addiotional option to the existing `./azure_vms.rb | tidal sync ` workflow

**Comment by dcolebatch** [comment]
Thanks @gkiranreddy!

1) Just confirming: you ran this and it worked ok?  Particularly the custom fields changes, did they come through as you expected?

2) with respect to the periodic saving to file- I believe the intent here was to be able to handle very large environments without running out of memory, or causing 500s on the Tidal API?

 I have two observations:
a) We appear to write to file, but also maintain the data in a global variable? (So, our memory footprint might actually be worse now)
b) we delete the file at the end, which would prevent us from being able to re-play the data to `tidal sync...` manually when troubleshooting.

I wonder if we might consider periodically writing to file *and* syncing to tidal tools if some flag is set?

e.g. the below, with no STDOUT writing of JSON:
```./azure_vms.rb -sync-page 10```

as an addiotional option to the existing `./azure_vms.rb | tidal sync ` workflow

**Comment by gkiranreddy** [comment]
Custom Fields, validated and yes they came through no issues there. 

Regarding writing to the file, I think I literally took the note of writing to the file :)
I understand and it makes sense to write to file and sync at the same time. 

I will try and update the script to take the arg, based on arg batch it, sync to tidal, also write to file. Will delete the files deletion lines.   

Please let me know if any comments.

**Comment by dcolebatch** [comment]
Sounds great @gkiranreddy

**Comment by gkiranreddy** [comment]
@dcolebatch updated the script 

- Updates based on the PR review 
- Batching feature as discussed 

we do not need to run the script `./azure_vms.rb | tidal sync servers` anymore. 
Just running the script `./azure_vms.rb` should work. 
```
$ ./azure_vms-working2.rb 
Fetching subscriptions...
=> Found 2 subscriptions.
=> Found 21 resource groups in subscription 4c1a8af4-85cb-44c7-9528-491d3848d341.
=> Found 1 VMs in resource group my-app-Baz6
=> Found 1 VMs in resource group my-app-1ZQb
=> Found 1 VMs in resource group my-app-4Uhb
=> Found 1 VMs in resource group my-app-E4ds
=> Found 1 VMs in resource group my-app-ZeO5
=> Found 1 VMs in resource group my-app-rvSC
=> Found 1 VMs in resource group my-app-2ErM
=> Found 1 VMs in resource group my-app-KkkI
=> Found 1 VMs in resource group my-app-6kFu
=> Found 1 VMs in resource group my-app-wCT4
=> Found 5 resource groups in subscription a6bd8225-08e3-445b-88e4-9279009ddf36.
=> Found 2 VMs in resource group rhel5-demo
Fetching subscriptions...
=> Found 2 subscriptions.
=> Found 21 resource groups in subscription 4c1a8af4-85cb-44c7-9528-491d3848d341.
=> Found 1 App Services in resource group soc-test-db
=> Found 5 resource groups in subscription a6bd8225-08e3-445b-88e4-9279009ddf36.
=> Found 1 App Services in resource group tidal-example-resource-group
Server Sync Successful!
14 entries were created/updated.
Synced 14 of 14 entries to Tidal portal.
Server Sync Successful!
14 entries were created/updated.
```

To run with batches: `./azure_vms-working2.rb -sync-page 5`
```
$ ./azure_vms-working2.rb -sync-page 5
Fetching subscriptions...
=> Found 2 subscriptions.
=> Found 21 resource groups in subscription 4c1a8af4-85cb-44c7-9528-491d3848d341.
=> Found 1 VMs in resource group my-app-Baz6
=> Found 1 VMs in resource group my-app-1ZQb
=> Found 1 VMs in resource group my-app-4Uhb
=> Found 1 VMs in resource group my-app-E4ds
=> Found 1 VMs in resource group my-app-ZeO5
=> Found 1 VMs in resource group my-app-rvSC
=> Found 1 VMs in resource group my-app-2ErM
=> Found 1 VMs in resource group my-app-KkkI
=> Found 1 VMs in resource group my-app-6kFu
=> Found 1 VMs in resource group my-app-wCT4
=> Found 5 resource groups in subscription a6bd8225-08e3-445b-88e4-9279009ddf36.
=> Found 2 VMs in resource group rhel5-demo
Fetching subscriptions...
=> Found 2 subscriptions.
=> Found 21 resource groups in subscription 4c1a8af4-85cb-44c7-9528-491d3848d341.
=> Found 1 App Services in resource group soc-test-db
=> Found 5 resource groups in subscription a6bd8225-08e3-445b-88e4-9279009ddf36.
=> Found 1 App Services in resource group tidal-example-resource-group
Server Sync Successful!
5 entries were created/updated.
Synced 5 of 14 entries to Tidal portal.
Server Sync Successful!
5 entries were created/updated.
Synced 10 of 14 entries to Tidal portal.
Server Sync Successful!
4 entries were created/updated.
Synced 14 of 14 entries to Tidal portal.
Server Sync Successful!
14 entries were created/updated.
```

with `-sync-page` it gives status on the console by incrementing the # `Synced 5 of 14 entries to Tidal portal`

Please review

**Review by dcolebatch — APPROVED** [review]
Really nice changes @gkiranreddy - ready to merge!

**Pull request merged**